torch_xla/csrc/runtime/pjrt_registry.cc

```diff
@@ -14,10 +14,16 @@
 namespace torch_xla {
 namespace runtime {
 
-std::unordered_map<std::string, std::string> pjrt_plugins_;
-
 namespace {
 
+struct PluginEntry {
+  std::string library_path;
+  absl::flat_hash_map<std::string, xla::PjRtValueType> create_options;
+  bool init_coordinator;
```

Comment:
I wonder if we should drop `init_coordinator` and instead always initialize the coordinator when the distributed env vars are set, even on TPU where it's not strictly necessary. As long as torchrun launches the training in a distributed context, the env vars should be set, which I believe covers all GPU use cases since we plan to use torchrun for GPU SPMD (cc @vanbasten23).

On TPU it's not currently required, but we can always detect the environment from the GCE metadata and set the env vars for the user in a distributed context, since we don't require torchrun for multicontroller execution.

Reply:
> I wonder if we should drop init_coordinator and instead always initialize the coordinator when the distributed env vars are set, even on TPU where it's not strictly necessary.

In this case, I think we still want to keep `requires_xla_coordinator` option. We would just throw an error immediately if we don't have enough information to start the coordinator.

My other idea initially was that we could ask the plugin for the master IP, local rank, global rank, and world size, perhaps just asking `torch.distributed` for those values in the default implementation.

Reply:
I see - that makes sense. Just for context why I brought this up, JAX recently started requiring the coordinator to be initialized before the backend can be used (even on TPUs), but I'm not sure on the reason.

Keeping `init_coordinator` sounds fine to me. Thanks Will!